File: ios/Records/TransitionShape.swift
import CoreGraphics

struct TransitionShape: Equatable {
  var blurIntensity: CGFloat = 1
  var maxBlurRadius: CGFloat = .greatestFiniteMagnitude
  var enterScale: CGFloat = Defaults.enterScale
  var travelRatio: CGFloat = Defaults.travelRatio

  enum Defaults {
    static let enterScale: CGFloat = 0.4
    static let travelRatio: CGFloat = 1.0 / 3.0
    static let blurIntensity: CGFloat = 1
  }
}

File: web/records/TransitionShape.ts
interface TransitionShape {
  blurIntensity: number;
  maxBlurRadius: number;
  enterScale: number;
  travelRatio: number;
}

const TRANSITION_SHAPE_DEFAULTS = {
  blurIntensity: 1,
  maxBlurRadius: Number.POSITIVE_INFINITY,
  enterScale: 0.4,
  travelRatio: 1 / 3,
} as const satisfies TransitionShape;

export { TRANSITION_SHAPE_DEFAULTS };
export type { TransitionShape };
